pbench_logparse/main.go

```diff
@@ -0,0 +1,317 @@
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/sjug/go-logparse/stats"
+)
+
+var searchDir, resultDir, processes string
+var processList []string
+var blockDevices, netDevices stringSlice
+var fileHeader = map[string][]string{}
+
+type stringSlice []string
+
+type host struct {
+	kind      string
+	resultDir string
+	results   []resultType
+}
+
+type resultType struct {
+	kind, path           string
+	min, max, avg, pct95 float64
+}
+
+func (s *stringSlice) String() string {
+	return fmt.Sprintf("%v", *s)
+}
+
+func (s *stringSlice) Set(value string) error {
+	*s = append(*s, value)
+	return nil
+}
+
+func initFlags() {
+	flag.StringVar(&searchDir, "i", "/var/lib/pbench-results/test1/tools-default/", "pbench run result directory to parse")
+	flag.StringVar(&resultDir, "o", "/tmp/", "directory to output parsed CSV result data")
+	flag.StringVar(&processes, "proc", "openshift_start_master_api,openshift_start_master_controll,openshift_start_node,/etcd", "list of processes to gather")
+	flag.Var(&blockDevices, "blkdev", "List of block devices")
```

Comment:
Any specific reason for not having a comma separated values here? In fact, if we combine all the input params into a struct, it could be passed along easily. WDYT?